### PR TITLE
docs(guides): fix HMR Gotcha section snippet (#1463)

### LINKED ISSUE #1461: Guides - Warnings/Errors in Hot Module Replacement
Hello! I am new to Webpack, and I'm trying to learn it by the guide, as I advanced to HMR, I got some errors...

I've written the code showed in that chapter until the Stylesheets part (which I haven't read yet), and I got this console message while using `npm start`:

```
ERROR in chunk print [entry]
bundle.js
Conflict: Multiple assets emit to the same filename bundle.js
```

I'm not sure, but I think this error is about having multiple entries, so I guess it needs to have a output name for each one of it.

Then I noticed that the author of the guide changed the `filename` property of `output` from `[name].bundle.js` to `bundle.js`, so I typed it back to the previous value to see what happens, then I get no error from the console. But also no hot edits work also...

When I edit(and save) the `print.js` file, I get the following browser's console:

image

The `ReferenceError` is understandable(and they need to be solved on the guide), but even getting through them we have the warnings, that tell us the modules can't be hot updated, and that's the focus of this section of the guide...

Does anybody who had read the article about HMR know how to proceed? I really don't have a clue of how to solve and improve it...

### PR DESCRIPTION
**Pull request by maheshpec**
Fix for snippet in the _Gotcha_ section of [_Hot Module Replacement Guide_](https://webpack.js.org/guides/hot-module-replacement/#gotchas).

Resolves #1461

### CODE REVIEW

**Comment by skipjack** [comment]
@maheshpec does this resolve the issue you opened? Also, I think your issue may overlap with #1461 so I may close yours in favor of the first but I will double check before doing so.

**Comment by skipjack** [comment]
See my comment above... I think we can make a few changes to put the confusion to bed and make the code snippets flow a bit better.

@sbaidon can you explain more what those other warnings mentioned in #1461 were about? You mentioned some change to the `entry` configuration fixed them?

**Comment by sbaidon** [comment]
Ok, so I re-organized the block as you mentioned. I am not getting any kind of error, but the warning is still showing up:

screen shot 2017-07-30 at 12 05 00 pm

As you can see the `print.js` module is being updated, but then the warning is telling us it wasn't ?

My guess was that somehow webpack was trying to update the module twice, and it is only being accepted the first time. 

I thought it had something to do with having both `print.js` and `index.js` listed as entry points in our configuration.

So when I removed `print.js` as an entry this is the output I got:

screen shot 2017-07-30 at 12 27 00 pm

By the way I added the `new webpack.NamedModulesPlugin()` to the config so we could see names instead of ids.

Still not sure if I am doing anything else wrong...

**Comment by maheshpec** [comment]
I think @sbaidon is right. It is trying to load it twice - once for the `index` and other for the `print` entry point. When its trying to load the `print` entry point, it is getting `unaccepted` in `getAffectedStuff` function within `hotApply` function of `HotModuleReplacement.runtime.js` file within webpack (I'm using 3.4.1). The exact condition is 

```javascript
if(module.hot._main) {                                                                                  
        return {                                                                                                           
          type: "unaccepted",                                                                                                        
          chain: chain,                                                                                                    
          moduleId: moduleId                                                                                               
        };                                                                                                                 
} 
```

IMHO the approach to fix would be to add a _Preparation_ section which removes the print entry point. Please let me know what you guys think

**Comment by maheshpec** [comment]
I made the changes that @sbaidon mentioned and the ones about ordering that @skipjack mentioned. I've tested this with 3.4.1 version of webpack and it is working without any console warnings for me.

**Comment by skipjack** [comment]
@maheshpec @sbaidon yes good catch and @maheshpec thank you for updating! 

I should've caught that earlier, you shouldn't have an entry file that's consumed by another module without using the `CommonsChunkPlugin` and even there it usually just makes sense for vendor code splitting -- I completely missed that that was going on in these examples. I do understand how it happened when trying to synchronize this with the previous guides though. If we wanted to split `print.js` out to a separate chunk in the current setup, then `import()` would be a better solution.

As soon as the build passes, I'll get this merged so we can close out #1461.

File: content/guides/hot-module-replacement.md [hunk 1]
```diff
@@ -94,6 +88,13 @@ __index.js__
   }
 
   document.body.appendChild(component());
++
```
Comment (sbaidon) [inline content/guides/hot-module-replacement.md]: There's just this extra plus sign left

Comment (skipjack) [inline content/guides/hot-module-replacement.md]: I think that's ok, it's just indicating that a new line was added in between the last line of the script and new content. That's how we've done it in the other guides as well.

Comment (skipjack) [inline content/guides/hot-module-replacement.md]: Ah or maybe you meant there's two plus signs? That's actually just a looking at a `diff` within a `diff` issue 😆  (there's really only one in the document).

Comment (maheshpec) [inline content/guides/hot-module-replacement.md]: It is a little confusing to view these diffs :)

Comment (skipjack) [inline content/guides/hot-module-replacement.md]: yeah like some kind diff inception lol

Comment (sbaidon) [inline content/guides/hot-module-replacement.md]: oh I did mean that the new line was added, but if that's how it has been done on the other guides is fine 👍

**Pull request merged**